Output xcodebuild log in xcodebuild action

xfreebird opened the pull request:
updated `xcodebuild` action to be able to generate `xcodebuild.log` or/and disable `xcpretty`

xfreebird commented:
HI @KrauseFx what do you think of this PR ?

KrauseFx commented:
Really sorry for the delay here, will review it now

KrauseFx commented on lib/fastlane/actions/xcodebuild.rb:
```diff
@@ -112,8 +120,13 @@ def self.run(params)
         end
 
+        # By default we put xcodebuild.log in the current folder
+        buildlog_path ||= "."
```
Have we been doing that up until now? How about we store it in `~/Library/Logs/xctest` by default?

xfreebird replied:
Thanks for reviewing ! 
My experience with various cli tools is that usually they create log files in the folder I run them.
I will modify that behaviour. 

As a note, creating all logs in ```~/Library/Logs/xctest ``` looks reasonable when there is one user running one ```fastlane``` at a time. But on a CI box where there are running several jobs at a time, that is not a solution hence the need to have it in the workspace of the job.

KrauseFx commented:
Looks good, I'd just like to change the default output directory.

xfreebird commented:
@KrauseFx  sorry for spam, but I can't detect these warnings locally.

KrauseFx commented:
What happens when you run `rubocop`?

xfreebird commented:
I get no offences.

```shell
$ rubocop
Inspecting 218 files
..........................................................................................................................................................................................................................

218 files inspected, no offenses detected
```

The interesting part is that the ```xcodebuild_spec.rb``` has ```rubocop:disable Style/SpaceAroundOperators```. hound somehow ignores that ?

xfreebird commented:
@KrauseFx it seems @houndci ignores the "# rubocop:disable all" from the ```xcodebuild.rb``` action, and a similar directive from ```xcodebuild_spec.rb```.

xfreebird commented:
@houndci was reporting on ```xcodebuild.rb``` and ```xcodebuild_spec.rb``` this:
```
Do not use Time.now.strftime without zone. Use one of Time.zone.now.strftime, Time.now.strftime.current, Time.now.strftime.in_time_zone, Time.now.strftime.utc, Time.now.strftime.getlocal, Time.now.strftime.iso8601, Time.now.strftime.jisx0301, Time.now.strftime.rfc3339, Time.now.strftime.to_i, Time.now.strftime.to_f instead.
```

while ```rubocop``` locally was not. I added that suppress @houndci warnings.

xfreebird commented:
@KrauseFx what do you think ? Should I make more changes ? If you can make @houndci create same warnings as ```rubocop``` I'm more than happy to remove the changes related to ```rubocop```.

xfreebird commented:
@KrauseFx should I make additional changes to this MR ?

xfreebird commented:
@dtrenz @KrauseFx when you have time please look over this MR.
It is about:
* save xcodebuild log to a file option
* full xcodebuild logs in output (no xcpretty option) in xcodebuild action

There have been several requests, also I find this useful (xcodebuild log file).

KrauseFx commented on lib/fastlane/actions/xcodebuild.rb:
```diff
@@ -1,4 +1,3 @@
-# rubocop:disable all
```
You can leave this line here, so that you can get rid of the `TimeZone` part of the config file

KrauseFx commented on .rubocop.yml:
```diff
@@ -29,8 +34,12 @@ AllCops:
+    - 'lib/fastlane/actions/xcodebuild.rb'
 
-
+Rails/TimeZone:
```
Please remove from here. If you want to keep it, add it to the `xcodebuild.rb` file instead.

xfreebird committed:
Updated xcodebuild action to be able to generate xcodebuild.log or/and disable xcpretty

xfreebird commented:
@KrauseFx thanks for reviewing it, I removed the "Rails" rule from rubocop and added back the "disable all" to xcodebuild.rb. Now it seems there are the rubocop issues from the master branch.

Cheers!

KrauseFx commented:
Looks good, thanks :+1:
